echo_server.py

```diff
@@ -21,10 +24,49 @@ def echo_server():
                 if len(response) < 32:
                     done = True
                 text_received = "{}{}".format(text_received, response)
-            conn.sendall(text_received)
+            if text_received:
+                response_user = parase_request(text_received)
+            conn.sendall(response_user)
     except KeyboardInterrupt:
        server_socket.close()
 
+def response_ok(uri):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: text/plain'
+    body = uri
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    response  =  [first_line, timestamp, header, size, body, end]
+    return end.join(response)
+
+def response_error(error_code, error_message):
+    first_line = 'HTTP/1.1 {} {}'.format(error_code, error_message)
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: text/plain'
+    body = '{} {}'.format(error_code, error_message)
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    error_response = [first_line, timestamp, header, size, body, end]
+    return end.join(error_response)
+
+def parase_request(client_request):
+    full_text = client_request.splitlines()
+    first_line = full_text[0].split(' ')
+    response = error_determination(first_line)
+    return response
```

Comment:
In the specification, this method is supposed to take the request and parse it.  It should **raise a python error** if there is a problem with the request method or protocol, and should return the URI.  You are taking the right argument in, but the rest of the specification is ignored.